cmd/controller.go

```diff
@@ -51,6 +51,7 @@ import (
 func init() {
 	controllerCmd.Flags().StringVar(&controllerWorkerProfile, "profile", "default", "worker profile to use on the node")
 	controllerCmd.Flags().BoolVar(&enableWorker, "enable-worker", false, "enable worker (default false)")
+	controllerCmd.Flags().BoolVar(&single, "single", false, "enable single node (implies --enable-worker, default false)")
```

Comment:
Shouldn't we then also remove the `--enable-worker` flag? What happens is `--enable-worker` is true, but `--single` is not?

I would also suggest to use `--enable-worker` as an alias to `--single`, to avoid breakage. See: https://github.com/k0sproject/k0s/blob/main/cmd/installController.go#L30

Reply:
I think it's not exactly the same. We can run few controllers with `enable-worker` to work the same way as vanilla k8s clusters, though I don't know if there are any practical reasons to do it

Reply:
I agree they are the same, that's why I don't see the reason in keeping both flags.

Reply:
We cannot remove `--enable-worker` flag. Without this flag you cannot expand to multiples nodes which acts as both controllers and workers. (think 3 nodes which acts as both controllers *and* workers)

The `--single` flag disables the things that makes it possible to expand the control plane with more controllers.